CONTRIBUTING.rst

```diff
@@ -0,0 +1,292 @@
+    When pushing a rebased PR branch, we'd have to build every commit on that
+    branch individually. The loophole can be avoided by creating narrowly
+    focused PRs with only one logical change and therefore one commit.
+    Therefore, we consider the creation of PRs with longer histories to be a
+    privilege of trusted, long-time contributors. See ``
+
+
+* If a commit resolves (or contributes to the resolution of) an issue, we
+  mention that issue at the end of the commit title::
+
+    Reticulate them splines for good measure (#123)
+
+  Note that we don't use Github resolution keywords like "fixes" or "resolves".
+  Any mention of those preceding an issue reference in a title would
+  automatically close the issue as soon as the commit lands on the default
+  branch. This is undesirable as we want to continue to track issues in
+  Zenhub's *Merged* and *Done* pipelines even after the commit lands on
+  `develop`.
+
+* We value `expressive and concise commit message titles`_ and we use Github's
+  limit on the length of a commit message title. A commit title is too long if
+  (and only if) Github truncates or wraps the title in its UI.
+
+* We don't use a period at the end of commit titles.
+
+* We use `sentence case`_ for commit titles.
+
+.. _expressive and concise commit message titles: https://chris.beams.io/posts/git-commit/
+
+.. _sentence case: https://utica.libguides.com/c.php?g=291672&p=1943001
+
+
+Issue Tracking
+==============
+
+* We use Github's builtin issue tracking and Zenhub.
+
+* We use `sentence case`_ for issue titles.
+
+* We don't use a period at the end of issue titles.
+
+* For issue titles we prefer brevity over precision or accuracy. Issue titles
+  are read many times and should be optimized toward quickly scanning them.
+  Potential omissions, inaccuracies and ambiguities in the title can be added,
+  corrected or clarified in the description.
+
+* We make liberal use of labels. Labels denoting the subject of an issue are
+  blue, those denoting the kind of issue are green, issues relating to the
+  development process are yellow. Important labels are red.
+
+* We prefer issue to be assigned to one person at a time. If the original
+  assignee needs the assistance by another team member, the issue should be
+  assigned to the assisting person. Once assistance was provided, the ticket
+  should be assigned back to the original assignee.
+
+
+Pull Requests
+=============
+
+* When naming PR branches we follow the template below::
+  
+    issues/$AUTHOR/$ISSUE_NUMBER-$DESCRIPTION
+      
+  ``AUTHOR`` is the Github profile name of the PR author.
+  
+  ``ISSUE_NUMBER`` is a numeric reference to the issue that this PR addresses.
+  
+  ``DESCRIPTION`` is a short (no more than nine words) slug_ describing the
+  branch
+  
+* We rebase PR branches daily but …
+
+* … we don't eagerly squash them. Changes that address the outcome of a review
```

Comment:
So this means I can't use `git commit -a --amend`? :disappointed:

Reply:
Yes, it's more transparent considering that we also rebase. By disallowing amended commits the reviewer can ignore force pushes and always have a track record of the changes made in response to their comments and how those affected the build status.